File: atlaxchange_ledger/models/atlaxchange_dashboard.py
from odoo import models, api

class AtlaxchangeLedgerDashboard(models.Model):
    _name = 'atlaxchange.ledger.dashboard'
    _description = 'Atlaxchange Ledger Dashboard'

    @api.model
    def get_dashboard_data(self):
        Ledger = self.env['atlaxchange.ledger']
        transactions = Ledger.search([])

        # Compute metrics
        total_transactions_count = len(transactions)
        successful_payouts = transactions.filtered(lambda t: t.status == 'success')
        pending_payouts = transactions.filtered(lambda t: t.status == 'pending')
        failed_payouts = transactions.filtered(lambda t: t.status == 'failed')

        return {
            'total_transactions_count': total_transactions_count,
            'total_successful_payouts_count': len(successful_payouts),
            'pending_payouts_count': len(pending_payouts),
            'failed_payouts_count': len(failed_payouts),
        }

File: atlaxchange_ledger/static/src/js/atlaxchange_dashboard.js
/** @odoo-module **/

import { Component, useState, onWillStart } from "@odoo/owl";
import { rpc } from "web.rpc";

export class AtlaxchangeDashboard extends Component {
    static template = "atlaxchange_ledger.AtlaxchangeDashboard";

    state = useState({
        totalTransactions: 0,
        successfulPayouts: 0,
        pendingPayouts: 0,
        failedPayouts: 0,
        loading: true,
        error: null,
    });

    async fetchDashboardData() {
        try {
            const result = await rpc.query({
                model: "atlaxchange.ledger.dashboard",
                method: "get_dashboard_data",
            });
            this.state.totalTransactions = result.total_transactions_count || 0;
            this.state.successfulPayouts = result.total_successful_payouts_count || 0;
            this.state.pendingPayouts = result.pending_payouts_count || 0;
            this.state.failedPayouts = result.failed_payouts_count || 0;
            this.state.loading = false;
        } catch (error) {
            console.error("Error fetching dashboard data:", error);
            this.state.error = "Failed to load dashboard data.";
            this.state.loading = false;
        }
    }

    setup() {
        onWillStart(() => this.fetchDashboardData());
    }
}